Scripts/RemediationScripts/Remediate-Azure_AppService_DP_Use_Secure_FTP_Deployment.ps1

```diff
@@ -0,0 +1,1132 @@
+            }
+
+        Write-Host "[Step 4 of 4] Enable Secure FTP Deployments for App Services"
+        Write-Host $([Constants]::SingleDashLine)
+        # To hold results from the remediation.
+        $appServicesRemediated = @()
+        $appServicesSkipped = @()
+        
+        if($FTPSOnly.IsPresent -eq $true)
+        {
+            $userInputforFTPState="1"
+        }
+        else
+        {
+            if($DisableFTP.IsPresent -eq $true)
+            {
+                $userInputforFTPState="2"
+            }
+        }
+        
+        $appServicesWithFTPStateAllAllowed | ForEach-Object {
+            $appService = $_
+            $resourceId = $_.ResourceID
+            $resourceGroupName = $_.ResourceGroupName
+            $resourceName = $_.ResourceName
+            $isFTPStateAllAllowedForProdSlot = $_.isFTPConfiguredOnProductionSlot
+            $isSecureFTPEnabledOnAllNonProdSlots = $_.isSecureFTPEnabledOnAllNonProdSlots
+            $nonProdSlotsWithFTPStateAllAllowed = $_.nonProdSlotsWithFTPStateAllAllowed
+
+            Write-Host "Updating FTP State for App Service: Resource ID: [$($resourceId)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $nonProdSlotsWithFTPStateAllAllowedStr = $nonProdSlotsWithFTPStateAllAllowed -join ','
+            $isFTPStateAllAllowedForProdSlotPostRemediation = $isFTPStateAllAllowedForProdSlot
+            
+            # Reset the status further down, as appropriate.
+            $appService | Add-Member -NotePropertyName NonProductionSlotsSkipped -NotePropertyValue $nonProdSlotsWithFTPStateAllAllowedStr
+            $appService | Add-Member -NotePropertyName isFTPConfiguredOnProductionSlotPostRemediation -NotePropertyValue $isFTPStateAllAllowedForProdSlotPostRemediation
+
+            # If FTP State is neither Disabled not  FTPS Only for the production slot
+            if (-not [System.Convert]::ToBoolean($isFTPStateAllAllowedForProdSlot))
+            {
+                try
+                {
+                    Write-Host "Configuring Secure FTP Deployment on the production slot..." -ForegroundColor $([Constants]::MessageType.Info)
+                    Write-Host $([Constants]::SingleDashLine)
+                    $resource = Get-AzWebApp -ResourceGroupName $resourceGroupName -Name $resourceName
+                    if($userInputforFTPState -eq "1"){
+                        $resource.SiteConfig.FtpsState = "FtpsOnly"
+                    }
+                    else{
+                        if($userInputforFTPState -eq "2"){
+                        $resource.SiteConfig.FtpsState = "Disabled"
+                        }
+                    }
+                    
+                    # Holding output of set command to avoid unnecessary logs.
+                    $temp = $resource | Set-AzWebApp -ErrorAction SilentlyContinue
```

Comment:
Set-AzWebApp command has direct paramter '-FtpsState' to set ftp state, that will save one extra call per slot/app.

Reply:
Directly used Set-AzWebApp command as mentioned above.